pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java

```diff
@@ -385,11 +391,38 @@ private static TransformStep newComputeFieldFunction(ComputeConfig config) {
     return ComputeStep.builder().fields(fieldList).build();
   }
 
+  @SneakyThrows
   private TransformStep newComputeAIEmbeddings(ComputeAIEmbeddingsConfig config) {
+    String targetSvc = config.getService();
+    if (targetSvc == null || Strings.isNullOrEmpty(targetSvc)) {
+      targetSvc = ComputeAIEmbeddingsConfig.SupportedServices.OPENAI.name();
+    }
+
+    ComputeAIEmbeddingsConfig.SupportedServices service =
+        ComputeAIEmbeddingsConfig.SupportedServices.valueOf(targetSvc.toUpperCase());
+
+    final EmbeddingsService embeddingService;
+    switch (service) {
+      case OPENAI:
+        embeddingService = new OpenAIEmbeddingsService(openAIClient, config.getModel());
+        break;
+      case HUGGINGFACES_REST:
+        String hfKey = System.getenv("huggingfaces.key");
```

Comment:
Let's configure "huggingFace" like we do for "openai", with a top level configuration element
Because that's the way users can set their token.

The topic "secrets" should not be stored in configuration is a separate topic that in our services is tackled in a way that we cannot discuss here